algopy_testing/src/algopy/__init__.py

```diff
@@ -1,5 +1,101 @@
 # ruff: noqa: PLC0414
-from algopy.primitives.biguint import BigUInt as BigUInt
-from algopy.primitives.bytes import Bytes as Bytes
-from algopy.primitives.string import String as String
-from algopy.primitives.uint64 import UInt64 as UInt64
+from collections.abc import Callable
+from typing import ParamSpec, TypeVar
+
+from algopy_testing.enums import OnCompleteAction, TransactionType
+from algopy_testing.models import Account, Application, Asset, Global, Itxn, Txn
+from algopy_testing.primitives import BigUInt, Bytes, String, UInt64
+
+
+# TODO: Refine further, currently simplified to limit the scope of state management abstractions PR
+class ARC4Contract:
+    pass
+
+
+# TODO: Refine further, currently simplified to limit the scope of state management abstractions PR
+class LocalState:
+    def __init__(
+        self,
+        type_: type[object],
+        /,
+        *,
+        key: bytes | str = "",
+        description: str = "",
+    ) -> None:
+        self.type_ = type_
+        self.key = key
+        self.description = description
+        self._state: dict[object, object] = {}
+
+    def __setitem__(self, key: object, value: object) -> None:
+        if isinstance(key, Account):
+            key = str(key)
+        elif isinstance(key, Application | Asset):
```

Comment:
This doesn't look right. local state allows indexing by UInt64 or int, which represents the foreign account index to use (so ultimately still indexing by Account). Any other types should be a TypeError

Reply:
@daniel-makerx good catch, added a validation to only accept account, int or uint64